File: packages/hmssdk_flutter/ios/Classes/Models/HMSRoleExtension.swift
//
//  HMSRoleExtension.swift
//  hmssdk_flutter
//
//  Copyright © 2021 100ms. All rights reserved.
//

import Foundation
import HMSSDK

class HMSRoleExtension {

    static func toDictionary(_ role: HMSRole) -> [String: Any] {
        [
            "name": role.name,
            "publish_settings": HMSPublishSettingsExtension.toDictionary(role.publishSettings),
            "subscribe_settings": HMSSubscribeSettingsExtension.toDictionary(role.subscribeSettings),
            "priority": role.priority,
            "permissions": HMSPermissionExtension.toDictionary(role.permissions)
        ]
    }
}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSRoleExtension.kt
package live.hms.hmssdk_flutter

import live.hms.hmssdk_flutter.hms_role_components.PermissionParamsExtension
import live.hms.hmssdk_flutter.hms_role_components.PublishParamsExtension
import live.hms.hmssdk_flutter.hms_role_components.SubscribeSettings
import live.hms.video.sdk.models.role.HMSRole

class HMSRoleExtension {
    companion object {
        fun toDictionary(role: HMSRole?): HashMap<String, Any?>? {
            val hashMap = HashMap<String, Any?>()
            if (role == null)return null

            hashMap["name"] = role.name
            hashMap["publish_settings"] = PublishParamsExtension.toDictionary(role.publishParams)
            hashMap["subscribe_settings"] = SubscribeSettings.toDictionary(role.subscribeParams)
            hashMap["priority"] = role.priority
            hashMap["permissions"] = PermissionParamsExtension.toDictionary(role.permission)

            return hashMap
        }
    }
}
